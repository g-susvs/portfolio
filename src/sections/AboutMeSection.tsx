import '../css/aboutme.css'

export const AboutMeSection = () => {
    return (
        <section id="aboutme" className="aboutme-section">
            <h3 className="aboutme__title subtitle">SOBRE MI</h3>
            <p className='aboutme__text'>
                Egresado de la carrera de <strong>Diseño y Desarrollo de Software</strong> en el instituto <strong>Certus</strong>, durante mi tiempo en el instituto he explorado diferentes lenguajes y frameworks, he conectado con otros programadores y explorando diferentes áreas del desarrollo de software.</p>
            <br />
            <p className='aboutme__text'>
                Como desarrollador, mi enfoque es crear sitios web completamente funcionales. Me especializo en el desarrollo frontend, pero también tengo experiencia en el desarrollo del lado del servidor.
            </p>
        </section>
    )
}
