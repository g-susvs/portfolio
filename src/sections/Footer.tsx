import { FC } from 'react'

export const Footer: FC = () => {
    return (
        <footer className='footer'>
            <div className='footer__body'>
                <p>
                    2024 - Creado con ❤️ por <strong>Jesús Valencia</strong>
                </p>
                <div className='buttons-group'>
                    <a target="blank" className="button button--linkedIn" href="https://www.linkedin.com/in/jesus-guillermo-valencia-salvador/">
                        <span className="icon-linkedin2"></span>

                    </a>
                    <a target="blank" className="button button--github" href="https://github.com/g-susvs">
                        <span className="icon-github"></span>

                    </a>
                </div>
            </div>
        </footer>
    )
}
