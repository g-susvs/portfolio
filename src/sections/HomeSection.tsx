import { FC, Suspense, lazy } from "react";
import { CustomButton } from "../components/CustomButton";

import '../css/home.css'
import { LoaderImage } from "../components/LoaderImage";

const Image = lazy(() => import('../components/LazyImage'))


export const HomeSection: FC = () => {
    return (
        <section id="home" className='home-section'>
            <figure className='img__container'>

                <Suspense fallback={<LoaderImage />}>
                    <Image src={"/assets/jesus.jpg"} alt={"Jesús Valencia"} width="220" height="220" />
                </Suspense>

            </figure>
            <div className="home__body">
                <p className="home__myName">Jesús Valencia</p>
                <h2 className="home__rolDev">{"<Desarrollador Web />"}</h2>
                <div className="buttons-group">
                    <CustomButton
                        customClass="button--linkedIn"
                        title="LinkedIn"
                        href="https://www.linkedin.com/in/jesus-guillermo-valencia-salvador/"
                    >
                        <span className="icon-linkedin2"></span>

                    </CustomButton>
                    <CustomButton
                        customClass="button--github"
                        title="Github"
                        href="https://github.com/g-susvs"
                    >
                        <span className="icon-github"></span>
                    </CustomButton>

                </div>

            </div>
        </section>
    )
}