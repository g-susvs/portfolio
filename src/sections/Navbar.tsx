import { FC, useState } from 'react';

export const Navbar: FC = () => {

    const [showMenu, setShowMenu] = useState(false)

    const handleClink = () => {
        setShowMenu(!showMenu)
    }

    return (
        <header className='navbar'>
            <nav className='navbar__container'>
                <h5 className='navbar__title'>Portafolio</h5>
                <div className={`navbarMenu ${(!showMenu) && 'navbarMenu--hidden'}`}>
                    <a href="#home" className='navbarMenu__item text-center' onClick={handleClink}>INICIO</a>
                    <a href="#aboutme" className='navbarMenu__item text-center' onClick={handleClink}>SOBRE MI</a>
                    <a href="#skills" className='navbarMenu__item text-center' onClick={handleClink}>HABILIDADES</a>
                    <a href="#projects" className='navbarMenu__item text-center' onClick={handleClink}>PROYECTOS</a>
                </div>
                <button className='buttonMenu' onClick={handleClink}>
                    <span className="icon-menu"></span>
                </button>
            </nav>

        </header>
    )
}
