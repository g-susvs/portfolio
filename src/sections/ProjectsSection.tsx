import { ProjectCard } from '../components/ProjectCard'
import { ProjectProps } from '../interfaces'

import '../css/projects.css'

const projects: ProjectProps[] = [
    {
        title: 'Notas Markdown App',
        imgPath: '/assets/projects/app-markdown.jpg',
        description: 'Aplicación diseñada para la toma de apuntes con formato Markdown, tomando inspiración de la versatilidad de Notion.',
        techs: ['React JS', 'Material UI', 'Node JS', 'Typescript'],
        urlDemo: 'https://notas-markdown.vercel.app/',
        urlRepository: 'https://github.com/g-susvs/notas-markdown',
    },
    {
        title: 'Chatbot Web',
        imgPath: '/assets/projects/chatbot.jpg',
        description: 'Proyecto académico, funcionalidad de un chatbot. Conexión a servidor con toda la lógica para manejar el input del usuario.',
        techs: ['React JS', 'CSS', 'Python'],
        urlDemo: 'https://chatbot-ui-g-susvs.vercel.app/',
        urlRepository: 'https://github.com/g-susvs/chatbot-ui',
    },
    {
        title: 'Todo App',
        imgPath: '/assets/projects/todoapp.jpg',
        description: 'Listar tareas pendientes y completadas.',
        techs: ['React JS', 'SASS'],
        urlDemo: 'https://todo-app-gsusvs.netlify.app/',
        urlRepository: 'https://github.com/g-susvs/To-Do-App',
    }
]

export const ProjectsSection = () => {
    return (
        <section id="projects" className="projects-section">

            <h3 className="projects__title subtitle">PROYECTOS</h3>
            <div className="projects__list">
                {
                    projects.map(project =>
                        <ProjectCard
                            key={project.title}
                            title={project.title}
                            imgPath={project.imgPath}
                            description={project.description}
                            techs={project.techs}
                            urlDemo={project.urlDemo}
                            urlRepository={project.urlRepository}
                        />)
                }
            </div>
        </section>
    )
}
