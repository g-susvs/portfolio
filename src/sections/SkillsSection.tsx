import { SkillCard } from '../components/SkillCard'
import { SkillProps } from '../interfaces'

import '../css/skills.css'

const skills: SkillProps[] = [
    {
        title: 'HTML',
        imgPath: '/assets/techs/html.svg'
    },
    {
        title: 'CSS',
        imgPath: '/assets/techs/css.svg'
    },
    {
        title: 'Javascript',
        imgPath: '/assets/techs/javascript.svg'
    },
    {
        title: 'Typescript',
        imgPath: '/assets/techs/typescript.svg'
    },
    {
        title: 'Sass',
        imgPath: '/assets/techs/sass.svg'
    },
    {
        title: 'Git',
        imgPath: '/assets/techs/git.svg'
    },
    {
        title: 'React JS',
        imgPath: '/assets/techs/react.svg'
    },
    {
        title: 'Node JS',
        imgPath: '/assets/techs/node.svg'
    },
    {
        title: 'MongoDB',
        imgPath: '/assets/techs/mongodb.svg'
    },
    {
        title: 'MySql',
        imgPath: '/assets/techs/mysql.svg'
    },
    {
        title: 'PostgreSQL',
        imgPath: '/assets/techs/postgres.svg'
    },
]

export const SkillsSection = () => {


    return (
        <section id="skills" className="skills-section">
            <h3 className="skills__title subtitle">HABILIDADES</h3>

            <div className="skills__list">
                {
                    skills.map(skill =>
                        <SkillCard
                            key={skill.title}
                            title={skill.title}
                            imgPath={skill.imgPath}
                        />)
                }
            </div>
        </section>
    )
}
