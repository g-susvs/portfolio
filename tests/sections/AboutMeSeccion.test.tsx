import { render } from "@testing-library/react"
import { AboutMeSection } from "../../src/sections/AboutMeSection"

describe('Test in <AboutMeSection/>', () => {

    test('should match snapshot', () => {

        const { container } = render(
            <AboutMeSection />
        )

        expect(container).toMatchSnapshot()
    })
})