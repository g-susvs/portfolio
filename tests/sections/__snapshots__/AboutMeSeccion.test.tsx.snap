// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AboutMeSection/> should match snapshot 1`] = `
<div>
  <section
    class="aboutme-section"
    id="aboutme"
  >
    <h3
      class="aboutme__title subtitle"
    >
      SOBRE MI
    </h3>
    <p
      class="aboutme__text"
    >
      Hola 游녦 mi nombre es Jes칰s Valencia, he culminado mis estudios de 
      <strong>
        Dise침o y Desarrollo de Software
      </strong>
       en el instituto 
      <strong>
        Certus
      </strong>
      . Durante mi tiempo en el instituto, he podido conocer las  diversas 치reas del desarrollo de software, estableciendo conexiones con compa침eros programadores, compartiendo ideas y desarrollando juntos.
    </p>
    <br />
    <p
      class="aboutme__text"
    >
      Como programador mi enfoque se orienta a la creaci칩n de sitios web funcionales, con experiencia desarrollando tanto frontend como backend.
    </p>
  </section>
</div>
`;
